src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -3449,7 +3449,8 @@ private void login(String primary, String primaryInstanceName, int primaryPortNu
                                                                                                  // packetsize, etc)
                         || (SQLServerException.ERROR_SOCKET_TIMEOUT == driverErrorCode // socket timeout
                                 && (!isDBMirroring || attemptNumber > 0)) // If mirroring, only close after failover has been tried (attempt >= 1)
-                        || timerHasExpired(timerExpire)
+                        || timerHasExpired(timerExpire) || (connectRetryCount == 0 && !useTnir)
+                        || (connectRetryCount == 0 && useTnir && attemptNumber >= 1)
```

Comment:
```suggestion
                        || timerHasExpired(timerExpire) 
                        || (connectRetryCount == 0 && !useTnir)
                        || (connectRetryCount == 0 && useTnir && attemptNumber >= 1)
```

Reply:
manual formatting will be overriden by formatter